cocos2d/shaders/CCGLStateCache.js

```diff
@@ -42,14 +38,71 @@ var MAX_ACTIVETEXTURE = 0,
     _uVAO = 0;
 if (ENABLE_GL_STATE_CACHE) {
     MAX_ACTIVETEXTURE = 16;
-
     _currentShaderProgram = -1;
     _currentBoundTexture = [-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1];
     _blendingSource = -1;
     _blendingDest = -1;
     _GLServerState = 0;
     if(macro.TEXTURE_ATLAS_USE_VAO)
         _uVAO = 0;
+
+    var _currBuffers = {};
+
+    WebGLRenderingContext.prototype.glBindBuffer = WebGLRenderingContext.prototype.bindBuffer;
+    WebGLRenderingContext.prototype.bindBuffer = function (target, buffer) {
```

Comment:
Rewrite gl.bindBuffer, it will return wether the buffer at target position is changed, if not, then enableVertexAttribArray and vertexAttribPointer is not needed, can check out usage in RendererWebGL's _batchRendering